cartoframes/io/carto.py

```diff
@@ -111,7 +111,8 @@ def to_carto(dataframe, table_name, credentials=None, if_exists='fail', geom_col
         raise ValueError('Wrong dataframe. You should provide a valid DataFrame instance.')
 
     if isinstance(dataframe, GeoDataFrame):
-        check_crs(dataframe)
+        if is_reprojection_needed(dataframe):
+            dataframe.to_crs(epsg=4326)
```

Comment:
Maybe we can create another utils function to reproject with `gdf.to_crs(epsg=4326)` so we can hide the magic numbers.

The method `to_crs` returns the result (does not work inplace), so this should be changed to something like:

```py
if is_reprojection_needed(dataframe):
    dataframe = reproject(dataframe)
```